pkg/ccl/streamingccl/streamingest/stream_ingestion_processor_test.go

```diff
@@ -476,3 +561,14 @@ func makeCheckpointEventCounter(
 		}
 	}
 }
+
+func markGenerationEventAsReceived(
```

Comment:
can we pass a func() as a parameter here instead of a channel? This will give us flexibility in the future to add stuff to that func.

Maybe also rename the function to `makeGenerationEventReceived`